app/javascript/schools/courses/components/curriculum_editor/CurriculumEditor__ContentBlockCreator.re

```diff
@@ -193,34 +228,149 @@ let button = (target, aboveContentBlock, send, addContentBlockCB, blockType) =>
   </label>;
 };
 
-let uploadFile =
-    (target, send, addContentBlockCB, isAboveContentBlock, formData) =>
-  Api.sendFormData(
-    "/school/targets/" ++ (target |> Target.id) ++ "/content_block",
-    formData,
-    json => {
-      Notification.success("Done!", "File uploaded successfully.");
-      let contentBlock = json |> ContentBlock.decode;
-      addContentBlockCB(contentBlock);
-      send(FinishSaving(isAboveContentBlock));
-    },
-    () => send(FailToUpload),
+let embedUrlRegexes = [|
+  [%bs.re "/https:\\/\\/.*slideshare\\.net/"],
+  [%bs.re "/https:\\/\\/.*vimeo\\.com/"],
+  [%bs.re "/https:\\/\\/.*youtube\\.com/"],
+  [%bs.re "/https:\\/\\/.*youtu\\.be/"],
+|];
+
+let validEmbedUrl = url =>
+  Belt.Array.some(embedUrlRegexes, regex => regex->Js.Re.test_(url));
+
+let handleCreateEmbedContentBlock =
+    (target, aboveContentBlock, url, send, addContentBlockCB) =>
+  if (url |> validEmbedUrl) {
+    send(ToggleSaving);
+
+    let aboveContentBlockId =
+      aboveContentBlock |> OptionUtils.map(ContentBlock.id);
+
+    let targetId = target |> Target.id;
+
+    CreateEmbedContentBlock.make(~targetId, ~aboveContentBlockId?, ~url, ())
+    |> GraphqlQuery.sendQuery
+    |> Js.Promise.then_(result =>
+         handleGraphqlCreateResponse(
+           aboveContentBlock,
+           send,
+           addContentBlockCB,
+           result##createEmbedContentBlock##contentBlock,
+         )
+       )
+    |> Js.Promise.catch(_ => {
+         send(FailedToCreate);
+         Js.Promise.resolve();
+       })
+    |> ignore;
+  } else {
+    send(
+      SetError(
+        "The URL doesn't look valid. Please make sure that it starts with 'https://' and that it's one of the accepted websites.",
+      ),
+    );
+  };
+
+let handleVimeoVideoUpload =
+    (file, vimeoVideo, send, target, aboveContentBlock, addContentBlockCB) => {
+  let url = vimeoVideo##uri;
+  let uploadUrl = vimeoVideo##uploadLink;
+
+  Tus.upload(
+    ~file=Tus.makeFile(file),
+    ~uploadUrl,
+    ~onError=
+      error => {
+        Js.log(error);
```

Comment:
Let's actually leave this in. We do this elsewhere in the codebase as well - this is the only way to figure out what error happened within the browser, if any occur during development.

What should actually be added is a `Rollbar.error` call so that production errors get logged.

Reply:
@harigopal should we add the `ErrorUtils.logUnexpectedError` (Notify and Report to Rollbar) helper to handle it?

Reply:
@bodhish Hmm. Not a bad idea, but it does have some issues. I think we can discuss that outside of this PR, and implement it later.